src/utils/scalloc.c

```diff
@@ -21,3 +22,29 @@ scalloc(size_t n, size_t s, char *file, size_t line)
     }
     return ret;
 }
+
+/*
+ * ssprintf
+ *      Allocates the exact amount of memory for the formatted string and
+ *      returns formatted string. Returned memory must be released manually
+ *      by calling SFREE or free.
+ */
+char *
+ssprintf(const char *fmt, char *file, size_t line, ...)
```

Comment:
why not just use `asprintf` and/or `vasprintf`? afaict, the functionality is identical.

Reply:
agree, i wasn't aware of those versions of `printf`

Reply:
i switched to `asprintf`, but this required also to enable gnu extensions by defining `_GNU_SOURCES` instead of `_DEFAULT_SOURCES`. Is it something we should worry about?